src/logger/mongoose_json_formatter.erl

```diff
@@ -0,0 +1,105 @@
+%%%-------------------------------------------------------------------
+%%% @copyright 2020, Erlang Solutions Ltd.
+%%% @doc
+%%% A module formatting logger reports into JSON objects.
+%%%
+%%% @end
+%%%-------------------------------------------------------------------
+-module(mongoose_json_formatter).
+
+%% API
+-export([format/2]).
+
+-spec format(logger:log_event(), logger:formatter_config()) -> unicode:chardata().
+format(E = #{msg := {report, #{label := {error_logger, _}, format := Format, args := Terms}}}, FConfig) ->
+    format(E#{msg := {report,
+                      #{unstructured_log =>
+                        unicode:characters_to_binary(io_lib:format(Format, Terms))}}},
+           FConfig);
+format(E = #{msg := {string, String}}, FConfig) ->
+    format(E#{msg := {report,
+                      #{unstructured_log =>
+                        unicode:characters_to_binary(io_lib:format(String, []))}}}, FConfig);
+format(E = #{msg := {report, Report}}, FConfig) when is_map(Report) ->
+    NewMap = process_metadata(E),
+    NewReport = maps:merge(Report, NewMap),
+    NewConfig = maps:merge(default_config(), config_correct_depth(FConfig)),
+    Formatted = format_item(NewReport, NewConfig),
+    unicode:characters_to_binary([jiffy:encode(Formatted), "\n"], utf8);
```

Comment:
I was wondering, why is this `unicode:characters_to_binary/2` needed here, together with that final `\n`?

Reply:
`unicode:characters_to_binary/2` was to add the final `\n`. It was there because it wasn't added by default and everything was logged into one looong line.

Reply:
Jiffy returns binary. it's safe to concat `\n` to unicode strings as is.

Before:
```
unicode:characters_to_binary([jiffy:encode(Formatted), "\n"], utf8);
```

After:

```
Bin = jiffy:encode(Formatted),
<<Bin, "\n">>.
```